### ISSUE TITLE
Infinite Initializing

### ISSUE THREAD

**Issue by cph30075** [tf2td/towerdefense#5]
Hi, I managed to get the TF2 Attributes plugin working after a bit of elbow grease and the rest _seem_ to work. After joining it just tells me to wait a moment while it initializes seemingly forever. I'm assuming the problem is the MariaDB server, as that is by far the most likely thing as I have absolutely no experience with databases. Thanks for any help.

Here's the log:
[l0617019.log](https://github.com/tf2td/towerdefense/files/3299467/l0617019.log)

Here's a screenshot of the HeidiSQL GUI of the DB:
![mysql](https://user-images.githubusercontent.com/36975359/59647768-79e26b80-914a-11e9-89e5-f329cc5378b4.png)

**Comment by HurpDurp** [comment]
Have you ensured the tower defense config is being executed? That also doesn't appear to be the full server log, you should see quite a bit of tower defense debug messages (sourcemod also doesn't look like it's loaded).

**Comment by cph30075** [comment]
I _think_ it's running thought I don't know how to verify that. I think I accidentally used the wrong location for the logs, should I be picking them from tf/addons/sourcemod/logs? Sourcemod is definitely running, I've tested by punching in "sm addons list"

**Comment by HurpDurp** [comment]
Are you seeing any console output from tower defense? There should be a `[TD]` prefix. `sm plugins list` should also show the tower defense plugin, and if it's loaded or not.

**Comment by cph30075** [comment]
Yeah, [TD] prefixes and the Tower Defense plugin is loaded.

**Comment by HurpDurp** [comment]
Can you grab all the td prefixed logs and paste them here?

**Comment by cph30075** [comment]
The logs in both Sourcemod and tf do not show any [TD] prefixed stuff, so I just started my server up and copy+pasted the logs starting from when the [TD] prefix showed up. Hope it's enough. After looking at it, I'm starting to think that it's failing at "Query failed at Database_AddServer..." but that's just me.

https://pastebin.com/bzLMqeqp

**Comment by HurpDurp** [comment]
Yep, that would indicate a database issue. Towerdefense was written for
MySQL and it looks like the schema isn't directly compatible with mariadb
-- you could try this configuration change here:
https://www.farbeyondcode.com/Solution-for-MariaDB-Field--xxx--doesn-t-have-a-default-value-5-2720.html

You might also consider switching to mysql as there might be a few other
mariadb quirks that prevent the plugin from loading.

**Comment by cph30075** [comment]
Yeah, looks like I'll have to switch to MySQL, I punched in the listed command into my.ini, but when I started the server it still wouldn't work. I'll report back my findings once I switch to MySQL

**Comment by cph30075** [comment]
It appears to _sorta_ work. The only issue(s) is that I can join any team and go any class with any weapon (i.e. I'm not normally restricted to just the stock shotty and pistol as engineer). I can also freely build a dispenser and a teleporter (don't remember if you normally can). No metal packs spawn and I cannot start the game; the red start button does not go away and the server does not acknowledge me shooting it at all.

**Comment by HurpDurp** [comment]
So the plugin works after the reload? What version of mysql server did you install?

**Comment by cph30075** [comment]
Most recent one via the Oracle download page here: https://dev.mysql.com/downloads/

Downloaded the community release for Windows 64 bit.

**Comment by HurpDurp** [comment]
Interesting... You could try an older MySQL version, otherwise I'll try
getting my development server back up and running next week and get back to
you.

It's possible that the newer version of MySQL is incompatible with the
sourcemod db driver, however that seems unlikely as I don't see any other
reports of it not working.

**Comment by cph30075** [comment]
Strange. I'll see if I can get an old version but it's a bit of a pain at this point. I'll report back if the older version works assuming I actually find the time to get it installed lol.

**Comment by rumblefrog** [comment]
The latest master schema versions seem to work fine for 10.1.40-MariaDB. Although missing the host table within the dump.

**Comment by cph30075** [comment]
I already switched from MariaDB to MySQL, but it still doesn't work - I'll report back once I insert your patch and see if anything else starts working.

**Comment by rumblefrog** [comment]
Sure, let me know if it works.

@HurpDurp , any chance you have the `host` table schema?

**Comment by HurpDurp** [comment]
I think I do - though I won't have access to my PC until Monday to check.
It shouldn't be too hard to rebuild in any case.

**Comment by HurpDurp** [comment]
I've updated the schema to remove references to the host table, it is no longer used since I removed the server verification code from the plugin a while back. I started up a new server with the schema and things were working once I reloaded the map once (it was stuck on initializing briefly).

**Comment by HurpDurp** [comment]
Also @rumblefrog,

If you want to submit that metal pack patch as a PR I'lll accept it.
